### ISSUE TITLE
'No such file or directory' in tmp folder

### ISSUE THREAD

**Issue by SilasPeters** [NixOS/nix#9062]
## Platform
Manjaro Manjaro Linux x86_64

## Additional information
I have installed nix before, but through ``pamac install nix``. I decided to remove it, to then install it through the multi-user command as specified on the website. Prior to this, I removed the complete /nix folder, and uninstalled it through ``pamac remove nix``. I believe that pamac installed the single-user nix variant.

## Output

<details><summary>Output</summary>

```log
Note: a multi-user installation is possible. See https://nixos.org/manual/nix/stable/installation/installing-binary.html#multi-user-installation
Switching to the Multi-user Installer
Welcome to the Multi-User Nix Installation

This installation tool will set up your computer with the Nix package
manager. This will happen in a few stages:

1. Make sure your computer doesn't already have Nix. If it does, I
   will show you instructions on how to clean up your old install.

2. Show you what I am going to install and where. Then I will ask
   if you are ready to continue.

3. Create the system users (uids [30001..30032]) and groups (gid 30000)
   that the Nix daemon uses to run builds.

4. Perform the basic installation of the Nix files daemon.

5. Configure your shell to import special Nix Profile files, so you
   can use Nix.

6. Start the Nix daemon.

Would you like to see a more detailed list of what I will do?
[y/n] n


---- let's talk about sudo -----------------------------------------------------
This script is going to call sudo a lot. Every time I do, it'll
output exactly what it'll do, and why.

Just like this:

---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo echo

to demonstrate how our sudo prompts look


This might look scary, but everything can be undone by running just a
few commands. I used to ask you to confirm each time sudo ran, but it
was too many times. Instead, I'll just ask you this one time:

Can I use sudo?
[y/n] y

Yay! Thanks! Let's get going!

~~> Checking for artifacts of previous installs
Before I try to install, I'll check for signs Nix already is or has
been installed on this system.

---- warning! ------------------------------------------------------------------
Nix already appears to be installed. This installer may run into issues.
If an error occurs, try manually uninstalling, then rerunning this script.


Uninstalling nix:
1. Delete the files Nix added to your system:

  sudo rm -rf "/etc/nix" "/nix" "/root/.nix-profile" "/root/.nix-defexpr" "/root/.nix-channels" "/root/.local/state/nix" "/root/.cache/nix" "/home/silas/.nix-profile" "/home/silas/.nix-defexpr" "/home/silas/.nix-channels" "/home/silas/.local/state/nix" "/home/silas/.cache/nix"

and that is it.


---- Nix config report ---------------------------------------------------------
        Temp Dir:       /tmp/tmp.8PgK4hjfv4
        Nix Root:       /nix
     Build Users:       32
  Build Group ID:       30000
Build Group Name:       nixbld

build users:
    Username:   UID
     nixbld1:   30001
     nixbld2:   30002
     nixbld3:   30003
     nixbld4:   30004
     nixbld5:   30005
     nixbld6:   30006
     nixbld7:   30007
     nixbld8:   30008
     nixbld9:   30009
     nixbld10:  30010
     nixbld11:  30011
     nixbld12:  30012
     nixbld13:  30013
     nixbld14:  30014
     nixbld15:  30015
     nixbld16:  30016
     nixbld17:  30017
     nixbld18:  30018
     nixbld19:  30019
     nixbld20:  30020
     nixbld21:  30021
     nixbld22:  30022
     nixbld23:  30023
     nixbld24:  30024
     nixbld25:  30025
     nixbld26:  30026
     nixbld27:  30027
     nixbld28:  30028
     nixbld29:  30029
     nixbld30:  30030
     nixbld31:  30031
     nixbld32:  30032

Ready to continue?
[y/n] y


~~> Setting up the build group nixbld
            Exists:     Yes

~~> Setting up the build user nixbld1
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 1
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld2
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 2
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld3
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 3
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld4
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 4
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld5
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 5
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld6
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 6
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld7
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 7
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld8
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 8
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld9
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 9
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld10
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 10
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld11
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 11
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld12
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 12
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld13
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 13
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld14
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 14
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld15
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 15
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld16
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 16
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld17
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 17
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld18
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 18
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld19
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 19
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld20
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 20
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld21
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 21
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld22
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 22
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld23
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 23
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld24
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 24
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld25
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 25
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld26
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 26
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld27
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 27
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld28
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 28
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld29
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 29
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld30
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 30
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld31
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 31
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the build user nixbld32
            Exists:     Yes
            Hidden:     Yes
    Home Directory:     /var/empty
              Note:     Nix build user 32
   Logins Disabled:     Yes
  Member of nixbld:     Yes
    PrimaryGroupID:     30000

~~> Setting up the basic directory structure

---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo /bin/chown -R root:nixbld /nix

to take root ownership of existing Nix store files


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo install -dv -m 0755 /nix /nix/var /nix/var/log /nix/var/log/nix /nix/var/log/nix/drvs /nix/var/nix /nix/var/nix/db /nix/var/nix/gcroots /nix/var/nix/profiles /nix/var/nix/temproots /nix/var/nix/userpool /nix/var/nix/daemon-socket /nix/var/nix/gcroots/per-user /nix/var/nix/profiles/per-user

to make the basic directory structure of Nix (part 1)


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo install -dv -g nixbld -m 1775 /nix/store

to make the basic directory structure of Nix (part 2)


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo install -dv -m 0555 /etc/nix

to place the default nix daemon configuration (part 1)


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo install -m 0664 /tmp/tmp.8PgK4hjfv4/.nix-channels /root/.nix-channels

to set up the default system channel (part 1)


~~> Installing Nix

---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo cp -RPp ./store/01pvlzh6kwi0yw2m29s277bmgrrmi0ld-libcpuid-0.6.3 ./store/0kmx26k1zk1gclfswwk0hbvkv6l2ahfx-aws-c-http-0.7.6 ./store/0lvgi2bnwi8cs0wx2242g1275kqx8bba-libssh2-1.11.0 ./store/2av8x1ipwfm5n8nrx1i18bygza6v8b6g-brotli-1.0.9-lib ./store/2nvsyg7qp6vixv5bpwvvafy9kxi6qklz-aws-c-sdkutils-0.1.9 ./store/3kk0nlrfq90qfljsrk23nnlxjj06djfl-editline-1.17.1 ./store/428ig60cc3n8q8kg0cmpws07a12ksrqx-aws-sdk-cpp-1.11.37 ./store/4563gldw8ibz76f1a3x69zq3a1vhdpz9-libidn2-2.3.4 ./store/46m4xx889wlhsdj72j38fnlyyvvvvbyb-glibc-2.37-8 ./store/4iabmjjq95069myjsrid8pk2ib3yz4nn-openssl-3.0.10 ./store/567zfi9026lp2q6v97vwn640rv6i3n4c-libunistring-1.1 ./store/5wwvy6bam5wa0w3cfbghazck90jyxaxn-aws-c-event-stream-0.2.20 ./store/69jpyha5zbll6ppqzhbihhp51lac1hrp-zlib-1.2.13 ./store/6yp0l9mdrwqkd1a33ps6ykk2mnlyzx3v-acl-2.3.1 ./store/7aw32rkwgbim37x6kgza7qvr35czjysa-nghttp2-1.51.0-lib ./store/7lci7yd3nqsn79j8zrz9nl7y4vpma078-aws-c-mqtt-0.8.8 ./store/81d13il7plchw65gz8y9ywcxrngq149c-gcc-12.2.0-libgcc ./store/843dqq10jdkalr2yazaz6drx334visrb-gcc-12.2.0-lib ./store/8fv91097mbh5049i9rglc73dx6kjg3qk-bash-5.2-p15 ./store/b19c558b7c21a72hxr6cq61jhzz3h8hq-libsodium-1.0.18 ./store/cq0j3c5kilkh3dgqn0v6g94kjwhcxd15-libarchive-3.6.2-lib ./store/cqvxiq4l98b7d1ra0sqj1p83cb8sk99z-nix-2.18.0 ./store/fm7a66v8l2rjmfass0i7d4fqllxzs09z-aws-c-cal-0.5.21 ./store/gwsf52ljpp3pi13fflmclmn71nwgds12-aws-c-compression-0.2.16 ./store/h9rmj5nqwm89aj8zlm0cf4fxbbxdd8zm-xz-5.4.3 ./store/hj1caixhxrldx5jrx7x7axlcghn3r82n-s2n-tls-1.3.43 ./store/jd99cyc0251p0i5y69w8mqjcai8mcq7h-xgcc-12.2.0-libgcc ./store/jj38aw52c2rddpknmi3ifmlx4ygn3flb-aws-c-common-0.8.15 ./store/jwvvq9nyfrjvj10pl533my6d3gpn9nq0-zstd-1.5.5 ./store/kaa2mgjp0zjg3s5s75rdi3fk90d4yxzd-libxml2-2.10.4 ./store/m63030yga75q2xhs6g016anlsr3afpc3-aws-c-auth-0.6.26 ./store/mrg3c5a3b0395nfzwikja0f1cn5hm8wc-busybox-static-x86_64-unknown-linux-musl-1.36.1 ./store/n12a68qch9s85k6ni4m4r4xxr8lwys1i-sqlite-3.41.2 ./store/n84ndc2g6gc5grsxd56drrvzr2hccc2a-boehm-gc-8.2.2 ./store/nnla37mnr2nimvysvvmpn25vip3qiglc-aws-c-s3-0.2.8 ./store/qp4q2wd52fs4q48gdmd5i2c04xp8j9cl-aws-checksums-0.1.14 ./store/s3l67dahg8lj8zppyq851bkn97f4i9r0-nlohmann_json-3.11.2 ./store/v0iir4ksr0vb1px00yd80hip9yy5p4hk-bzip2-1.0.8 ./store/v4r4x88y91nc72sjbi9791hqsziiz4ns-aws-crt-cpp-0.19.8 ./store/whkpnh2db07rn1b7hcvfjfwag8imygvp-attr-2.5.1 ./store/wlacmv33nsrf2vnfa2akcym13zh6zzrx-libkrb5-1.20.1 ./store/xqblj2djqj4as6s5i00rkybcm4hmfldr-keyutils-1.6.3-lib ./store/y2qmx9f9bskxks88l3h6klhzxxwb10ca-curl-8.1.1 ./store/ykg2prfglnifc947a9n44py79mlyqmhw-aws-c-io-0.13.19 ./store/ys6g8wvr5rwxyfqrlvwd7vipj9ysqn70-nss-cacert-3.92 ./store/zp5f1mi3diiqx68c2cnca4cv1zchw6v4-libseccomp-2.5.4-lib /nix/store/

to copy the basic Nix files to the new store at /nix/store


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo chmod -R ugo-w /nix/store/

to make the new store non-writable at /nix/store

      Alright! We have our first nix at /nix/store/cqvxiq4l98b7d1ra0sqj1p83cb8sk99z-nix-2.18.0

---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo HOME=/root /nix/store/cqvxiq4l98b7d1ra0sqj1p83cb8sk99z-nix-2.18.0/bin/nix-store --load-db

to load data for the first time in to the Nix Database

      Just finished getting the nix database ready.

~~> Setting up shell profiles: /etc/bashrc /etc/profile.d/nix.sh /etc/zshrc /etc/bash.bashrc /etc/zsh/zshrc

---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo cp /etc/bashrc /etc/bashrc.backup-before-nix

to back up your current /etc/bashrc to /etc/bashrc.backup-before-nix


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo tee -a /etc/bashrc

extend your /etc/bashrc with nix-daemon settings


# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo cp /etc/profile.d/nix.sh /etc/profile.d/nix.sh.backup-before-nix

to back up your current /etc/profile.d/nix.sh to /etc/profile.d/nix.sh.backup-before-nix


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo tee -a /etc/profile.d/nix.sh

extend your /etc/profile.d/nix.sh with nix-daemon settings


# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo cp /etc/zshrc /etc/zshrc.backup-before-nix

to back up your current /etc/zshrc to /etc/zshrc.backup-before-nix


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo tee -a /etc/zshrc

extend your /etc/zshrc with nix-daemon settings


# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo cp /etc/bash.bashrc /etc/bash.bashrc.backup-before-nix

to back up your current /etc/bash.bashrc to /etc/bash.bashrc.backup-before-nix


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo tee -a /etc/bash.bashrc

extend your /etc/bash.bashrc with nix-daemon settings


# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo cp /etc/zsh/zshrc /etc/zsh/zshrc.backup-before-nix

to back up your current /etc/zsh/zshrc to /etc/zsh/zshrc.backup-before-nix


---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo tee -a /etc/zsh/zshrc

extend your /etc/zsh/zshrc with nix-daemon settings


# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix


~~> Setting up shell profiles for Fish with conf.d/nix.fish inside /etc/fish /usr/local/etc/fish /opt/homebrew/etc/fish /opt/local/etc/fish

~~> Setting up the default profile

---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo HOME=/root /nix/store/cqvxiq4l98b7d1ra0sqj1p83cb8sk99z-nix-2.18.0/bin/nix-env -i /nix/store/cqvxiq4l98b7d1ra0sqj1p83cb8sk99z-nix-2.18.0

to install a bootstrapping Nix in to the default profile

replacing old 'nix-2.18.0'
installing 'nix-2.18.0'
/usr/share/rvm/scripts/functions/environment: line 267: rvm_bash_nounset: unbound variable

---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo HOME=/root NIX_SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt /nix/store/cqvxiq4l98b7d1ra0sqj1p83cb8sk99z-nix-2.18.0/bin/nix-channel --update nixpkgs

to update the default channel in the default profile

unpacking channels...
/tmp/nix-binary-tarball-unpack.37pzTxth10/unpack/nix-2.18.0-x86_64-linux/install-multi-user: line 948: /tmp/tmp.8PgK4hjfv4/nix.conf: No such file or directory

---- oh no! --------------------------------------------------------------------
Oh no, something went wrong. If you can take all the output and open
an issue, we'd love to fix the problem so nobody else has this issue.

:(

We'd love to help if you need it.

You can open an issue at
https://github.com/NixOS/nix/issues/new?labels=installer&template=installer.md

Or get in touch with the community: https://nixos.org/community
```

</details>

## Priorities

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by abathur** [comment]
I think this is the actual culprit:

```
/usr/share/rvm/scripts/functions/environment: line 267: rvm_bash_nounset: unbound variable
```

It sounds like maybe there's a shell running down in here that is managing to pick up some rvm scripts from your environment. The installer's using nounset. I'm not sure off the top of my head why it's popping at this specific place, though, nor at what level the shell is running here. 

(It doesn't look like the installer itself does this. It might be down inside the `sudo HOME=/root /nix/store/cqvxiq4l98b7d1ra0sqj1p83cb8sk99z-nix-2.18.0/bin/nix-env -i /nix/store/cqvxiq4l98b7d1ra0sqj1p83cb8sk99z-nix-2.18.0` invocation, but that's a bit of a guess.)

**Comment by SilasPeters** [comment]
I am not that experienced with rvm setups, but I believe that my setup is a little wonky. As far as I know, the only thing that happens is that everytime I open a shell a little script is run that monitors current directory for ruby files.

Perhaps this behaviour is conflicting in general, but more probably I made a whoopsie somewhere.

I don't know how to fix it, but consider this issue an edge-case.

**Comment by abathur** [comment]
I'm not really sure, either. There may well be something here that Nix or the installer could/should be defending against.

What does `declare -p ENV BASH_ENV` output?

**Comment by SilasPeters** [comment]
``declare -p ENV BASH_ENV`` outputs the following, with exitcode 1:

```
declare: no such variable: ENV
declare: no such variable: BASH_ENV
```

Output is not different when running in bash or zsh.

**Comment by SilasPeters** [comment]
Solved it!
As mentioned, the rvm scripts ruined my experience. In my case, I had the following hook in my .profile (which is sourced in every shell):
``source /usr/share/rvm/src/rvm/scripts/rvm``

I commented this out. Last time I tried this, this did not work. Then I remembered, **I should log out and in again**. Now it worked!

TLDR; remove the rvm sourcing in your shell.
